src/components/OSCALMetadata.js

```diff
@@ -38,17 +53,215 @@ const OSCALMetadataPartiesCard = styled(Paper)(({ theme }) => ({
   backgroundColor: theme.palette.background.paper,
   position: "relative",
   overflow: "auto",
-  maxHeight: "12em",
+  maxHeight: "18em",
 }));
 
 // Returns a string with a locality-sensitive representation of this date
 const formatDate = (isoDate) => new Date(isoDate).toLocaleString();
 
+const TELEPHONE_ICON_MAPPING = {
+  home: <HomeIcon />,
+  mobile: <SmartphoneIcon />,
+  office: <BusinessIcon />,
+};
+
+const ADDRESS_ICON_MAPPING = {
+  home: <HomeIcon />,
+  work: <BusinessIcon />,
+};
+
+const UNKNOWN_TYPE_ICON = <HelpOutlineIcon />;
+
+function getIconOfType(mapping, contactType) {
+  if (!mapping || !contactType) {
+    return UNKNOWN_TYPE_ICON;
+  }
+  return mapping[contactType] ?? UNKNOWN_TYPE_ICON;
+}
+
+function getPhoneIcon(contactType) {
+  return getIconOfType(TELEPHONE_ICON_MAPPING, contactType);
+}
+
+function getAddressIcon(contactType) {
+  return getIconOfType(ADDRESS_ICON_MAPPING, contactType);
+}
+
+export function OSCALMetadataPartyAddress(props) {
+  const { address } = props;
+  const addrString = [
+    ...address["addr-lines"],
+    `${address.city}, ${address.state} ${address["postal-code"]}`,
+    address.country,
+  ]
+    .filter((line) => line)
+    .flatMap((item) => [item, <br key={item} />]);
+
+  return (
+    <Grid container spacing={1}>
+      <Grid item>{getAddressIcon(address.type)}</Grid>
+      <Grid item>
+        <Typography>{addrString}</Typography>
+      </Grid>
+    </Grid>
+  );
+}
+
+export function OSCALMetadataPartyEmail(props) {
+  return (
+    <Typography>
+      <Link href={`mailto:${props.email}`}>{props.email}</Link>
+    </Typography>
+  );
+}
+
+export function OSCALMetadataPartyTelephone(props) {
+  const { telephone } = props;
+  return (
+    <Typography>
+      {getPhoneIcon(telephone.type)}
+      <Link href={`tel:${telephone.number}`}>{telephone.number}</Link>
+    </Typography>
+  );
+}
+
+export function OSCALMetadataPartyDialog(props) {
+  const [open, setOpen] = React.useState(false);
+
+  const handleOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const PartyInfoTypes = {
+    address: "address",
+    telephone: "telephone",
+    email: "email",
+  };
+
+  const TYPE_MAPPING = (infoProps) => ({
+    address: <OSCALMetadataPartyAddress address={infoProps} />,
+    telephone: <OSCALMetadataPartyTelephone telephone={infoProps} />,
+    email: <OSCALMetadataPartyEmail email={infoProps} />,
+  });
+
+  const getPartyInfoList = (
+    list,
+    partyInfoType,
+    emptyMessage = "No information provided"
+  ) => {
+    if (!list?.length) {
+      return <Typography> {emptyMessage} </Typography>;
+    }
+
+    return list.map((item) => (
+      <ListItem
+        key={partyInfoType === "email" ? item : `${item?.type}--${item?.name}`}
+      >
+        {TYPE_MAPPING(item)[partyInfoType]}
+      </ListItem>
+    ));
+  };
+
+  return (
+    <CardActions>
+      <Button size="small" onClick={handleOpen}>
```

Comment:
```suggestion
      <Button size="small" variant="outlined" onClick={handleOpen}>
```

To give the user better indication of this being a button, we could also add an outline to it as well.

Reply:
Also maybe more icons? https://mui.com/material-ui/material-icons/?query=contactpage&selected=ContactPage

Reply:
This has been implemented in the latest revision. Buttons to open the contact modal are outlined